src/legacy/index.ts

```diff
@@ -59,16 +59,21 @@ async function depTreeToGraph(depTree: DepTree, pkgManagerName: string): Promise
   return shortenNodeIds(depGraph as types.DepGraphInternal, eventLoopSpinner);
 }
 
+// While constructing the graph, we form node IDs using hashes of the subtrees.
+// This ensures deduplication: identical tree branches will be represented by a single graph path.
```

Comment:
This comment is not so correct. more correct to say: "identical sub-trees will be represented by a single sub-graph"